src/utils/quickPickFile.ts

```diff
@@ -34,14 +34,36 @@ function createFileItem(rootFolder: vscode.WorkspaceFolder, uri: vscode.Uri): It
     };
 }
 
-async function resolveFileItemsInternal(rootFolder: vscode.WorkspaceFolder, fileUri: vscode.Uri | undefined, globPatterns: string[], message: string, canPickMany: boolean)
-    : Promise<Item[] | undefined> {
-    if (fileUri) {
-        return [createFileItem(rootFolder, fileUri)];
+function getDockerFileGlobPatterns(): string[] {
+    return getGlobPatterns([DOCKERFILE_GLOB_PATTERN], /^dockerfile$/i);
+}
+
+function getDockerComposeFileGlobPatterns(): string[] {
+    return getGlobPatterns([COMPOSE_FILE_GLOB_PATTERN], /^yaml$/i);
+}
+
+function getGlobPatterns(globPatterns: string[], fileTypeRegEx: RegExp): string[] {
+    const result: string[] = globPatterns;
+    try {
+        const config = vscode.workspace.getConfiguration('files').get<{}>('associations');
+        if (config) {
+            for (const globPattern of Object.keys(config)) {
+                const fileType = <string | undefined>config[globPattern];
+                if (fileType && fileTypeRegEx.test(fileType)) {
```

Comment:
`files.associations` use language IDs, so we can look for exactly `dockerfile` and `yaml`, rather than a regex test.

Reply:
I am following the existing pattern which checks for dockerfile or Dockerfile. I believe the language id can have alias we define Dockerfile as alias for dockerfile.

Reply:
I see, fair enough.